storage/innobase/handler/handler0alter.cc

```diff
@@ -8433,6 +8433,178 @@ innobase_update_foreign_cache(
 	DBUG_RETURN(err);
 }
 
+/** Changes SYS_COLUMNS.PRTYPE for one column.
+@param[in,out]	trx	transaction
+@param[in]	tableid	table ID as in SYS_TABLES
+@param[in]	name	table name
+@param[in]	prtype	new precise type
+@return		boolean flag
+@retval	true	on failure
+@retval false	on success */
+static
+bool
+innobase_change_field_versioning_try(
+	trx_t* trx,
+	const ulint tableid,
+	const char* name,
+	ulint prtype)
+{
+	DBUG_ENTER("innobase_change_field_versioning_try");
+
+	pars_info_t* info = pars_info_create();
+
+	pars_info_add_int4_literal(info, "prtype", prtype);
+	pars_info_add_ull_literal(info,"tableid", tableid);
+	pars_info_add_str_literal(info, "name", name);
+
+	dberr_t error = que_eval_sql(info,
+				     "PROCEDURE CHANGE_COLUMN_MTYPE () IS\n"
+				     "BEGIN\n"
+				     "UPDATE SYS_COLUMNS SET PRTYPE=:prtype\n"
+				     "WHERE TABLE_ID=:tableid AND NAME=:name;\n"
+				     "END;\n",
+				     false, trx);
```

Comment:
I would use `POS` instead of `NAME` in the `WHERE` condition, because the primary key of `SYS_COLUMNS` is `(TABLE_ID,POS)`.

Ideally we should check that exactly one record was updated.